third_party/move/move-compiler-v2/tests/checking-lang-v1/eq_inline_typed.move

```diff
@@ -0,0 +1,13 @@
+module 0x42::m {
+
+    inline fun foo(f: |&u64|) {
+    }
+
+    fun g() {
+        foo(|v: &u64| {
```

Comment:
Once this PR is in, we should fix the text in the move book: https://aptos.dev/en/build/smart-contracts/book/functions#current-restrictions, where we currently say this is not allowed.

Reply:
@brmataptos Just double checking this comment was seen.